### PR TITLE
fix(search): don't update widgets if props hasn't changed

### PR DESCRIPTION
**Pull request by mthuret**

### CODE REVIEW

**Comment by vvo** [comment]
Can you remove the commit about docs on multiple instances?

**Comment by mthuret** [comment]
@vvo: I was on it ;)

**Comment by vvo** [comment]
Could you walk us briefly around this bugfix? When what this happening?

**Comment by mthuret** [comment]
Bug: when managing the state of InstantSearch externally, several identical queries were sent to Algolia. 

Root cause: The componentWillReceiveProps hook of React gets called for both the InstantSearch root component and the widgets. But since we don't check for the props equality we're doing a new search even if their props are still the same.

File: packages/react-instantsearch/src/core/createConnector.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default function createConnector(connectorDesc) {
         props: this.getProps(nextProps),
       });
 
-      if (isWidget) {
+      if (isWidget && !shallowEqual(this.props, nextProps)) {
```
Comment (vvo) [inline packages/react-instantsearch/src/core/createConnector.js]: Right before we update the state, do we need to update the state if props did not change?

**Pull request merged**